R/utils.R

```diff
@@ -216,3 +216,14 @@ set_lang <- function(new_lang) {
 reset_lang <- function(old_lang) {
   if (is.na(old_lang)) Sys.unsetenv("LANGUAGE") else Sys.setenv(LANGUAGE = old_lang)
 }
+
+#' Create a \code{linter} closure
+#' @param fun A function that takes a source file and returns \code{lint} objects.
+#' @return The same function with its class set to 'linter'.
+#' @export
+Linter <- function(fun) { # nolint: object_name_linter.
+  if (!is.function(fun) || length(formals(args(fun))) != 1L) {
+    stop("`fun` must be a function of one argument.", call. = FALSE)
```

Comment:
WDYT about the error message? I first thought of "`fun` must be a monadic function." but I'm not sure this is universally understood.

Reply:
yea i wouldn't use such a CS term :smile: 

I tweaked it a bit, think it's fine now